packages/starknet-snap/src/extractPublicKey.ts

```diff
@@ -26,15 +26,13 @@ export async function extractPublicKey(params: ApiParams) {
       throw new Error(`The given user address is invalid: ${requestParamsObj.userAddress}`);
     }
 
-    if (await isUpgradeRequired(network, userAddress)) {
-      throw new Error('Upgrade required');
-    }
+    const { publicKey } = await getKeysFromAddress(keyDeriver, network, state, userAddress);
+    await validateAccountRequireUpgradeOrDeploy(network, userAddress, publicKey);
 
     let userPublicKey;
     const accContract = getAccount(state, userAddress, network.chainId);
     if (!accContract?.publicKey || num.toBigInt(accContract.publicKey) === constants.ZERO) {
```

Comment:
that logic is redundant, which getKeysFromAddress is doing the same 
but lets keep it here and mark to remove it later